### ISSUE TITLE
🚀 Feature Request: Support for vue

### ISSUE THREAD

**Issue by dbssman** [CopilotKit/CopilotKit#1169]
### 🎤 Tell us your idea

What is your idea?
Writing a wrapper or a library for vue

Do you have any use case(s) in mind?
Vue developers being able to use this product

Who would find this useful?
The Vue community, also while doing it, the project might find itself porting a lot of stuff to a framework agnostic solution, which would help a lot in writing wrappers for other frameworks

**Comment by NathanTarbert** [comment]
Hi @dbssman, Vue isn't on the short term roadmap, but someone from our community published a library to assist in integrating Vue with CopilotKit, and the last commit was fairly recent.  
I have not personally tried it, but it might be worth a look. 
Check it out here: https://github.com/fe-51shebao/vue-copilotkit

Let me know if this is helpful

**Comment by enekesabel** [comment]
Hey everyone!

I'm Abel, a FE developer working at https://fynk.com

We're heavily working on integrating agentic features into our application, and wanted to use CopilotKit as it comes with a nice set of convenience features + baked in chat ui out of the box.Thanks for creating the whole kit, you rock! 🚀
The only downside is that our FE is built using Vue, and it looks like supporting Vue is not on your roadmap:
https://github.com/CopilotKit/CopilotKit/issues/1169

I checked out your OSS repo, it looked well-structured, properly covered with tests, Stories + Demo included, so I decided to be bold and went on a side quest with Codex, creating a full Vue port.

I gave it strict instructions to mirror the React implementation, aiming for:
- Public API parity, with React concepts translated into Vue counterparts where it made sense
- Unit/component test parity
- Story parity
- Demo app parity
- Visual parity in Stories and the Demo app, with only very small differences (~0.5-1%)

I’ll be honest: I don’t know CopilotKit by heart, and I’m not deeply familiar yet with the source code, the full feature set, or your way of working and contributing to the project.

At this point, all Stories and the Demo app are working properly, so the next step for me is to start using the port internally at fynk.

We’d be totally open to pushing this work back upstream if that’s the kind of contribution you’d value. I mainly wanted to reach out, let you know what I’ve been working on, where things currently stand, and ask what the next steps would be if you’re open to it.

I’d also value your guidance on what parts I should double-check before proposing anything upstream. For example:
- what milestones or quality gates you’d expect before introducing a new framework package
- what areas tend to be easy to miss
- and what your expectations would be around maintenance, test coverage, docs, release readiness, and long-term parity

Repo:
https://github.com/fynk-com/CopilotKit

Cheers,
Abel

Ps.:
I tried contacting you on Discord in your `contributing` channel, but AutoMod kept blocking my message. Please let me know where & in what format should we continue this discussion if you're interested!

**Comment by jpr5** [comment]
Vue support is not on the roadmap. CopilotKit is React-focused. Closing as out of scope.

**Comment by adamreisnz** [comment]
@jpr5 Given you've recently shipped an Angular client, are there any plans to also ship a Vue client in the future?

**Comment by enekesabel** [comment]
@adamreisnz https://www.npmjs.com/package/@copilotkit/vue

Still got some rough edges, but works so far :) 

Check https://github.com/CopilotKit/CopilotKit/pull/5176 for more recent/updated docs - current README.md is not that helpful)

**Comment by NathanTarbert** [comment]
@adamreisnz, we appreciate your inquiry. Vue was just shipped very recently by @enekesabel and his team, in fact. 

It's being polished, and we are working to get the documentation up very soon.

**Comment by adamreisnz** [comment]
That's great, thanks team! We'll take it for a spin.

**Comment by NathanTarbert** [comment]
Would love to get your feedback when you have a chance. 

Would you be open to a short call? 
I would love to get some detailed notes on good and bad.

**Comment by adamreisnz** [comment]
@NathanTarbert yes I'd be happy to. I can follow up with you once we've started the migration (currently using React islands for CPK in our Vue app)
